diff -Naur libupnp_old/upnp/src/api/upnpapi.c libupnp_new/upnp/src/api/upnpapi.c
--- libupnp_old/upnp/src/api/upnpapi.c	2017-05-16 13:38:08.312212287 +0800
+++ libupnp_new/upnp/src/api/upnpapi.c	2017-05-16 13:37:54.984212311 +0800
@@ -3944,6 +3944,10 @@
 		memset(ifReq.ifr_name, 0, sizeof(ifReq.ifr_name));
 		strncpy(ifReq.ifr_name, pifReq->ifr_name,
 			sizeof(ifReq.ifr_name) - 1);
+		/* Skip USB over lan interface */
+		if(!strncmp(ifReq.ifr_name,"usb",3)) {
+			continue;
+		}
 		if (ioctl(LocalSock, SIOCGIFFLAGS, &ifReq) < 0) {
 			UpnpPrintf(UPNP_ALL, API, __FILE__, __LINE__,
 				"Can't get interface flags for %s:\n",
